### PR TITLE
Correct issue in 68000 sleigh

### PR DESCRIPTION
**Pull request by garrettgu10**
This should address Issue #4183

### CODE REVIEW

File: Ghidra/Processors/68000/data/languages/68000.sinc [hunk 1]
```diff
@@ -1719,7 +1719,7 @@ ptestLevel: "#"^mregn	is mregn  { export *[const]:1 mregn; }
 
 :rts				is opbig=0x4e & op37=14 & op02=5				{ PC = *SP; SP = SP+4; return [PC]; }
 
-:sbcd Tyb,Txb			is op=8 & op48=16 & Txb & Tyb
+:sbcd Txb,Tyb			is op=8 & op48=16 & Txb & Tyb
```
Comment (mumbel) [inline Ghidra/Processors/68000/data/languages/68000.sinc:1722]: Disassemble an instruction in ghidra and objdump.  Is display the same?  All this does is break the display right?

Comment (garrettgu10) [reply Ghidra/Processors/68000/data/languages/68000.sinc:1722]: oh wait, this is true

Comment (garrettgu10) [reply Ghidra/Processors/68000/data/languages/68000.sinc]: Sorry, I'm kinda rusty with SLEIGH. Also, it seems Txb corresponds to Dy in the manual and vice versa (?). Does this new modification make sense?

**Comment by flk0** [comment]
I'd like to provide my perspective on the root cause of this problem. As you've indicated, there is an inconsistent use of Dx/Dy in the reference manual, and it seems the SLEIGH spec was written assuming consistency. For instructions CMPM/ABCD/EXG/ADDX the register derived from bits 9-11 of the instruction is labelled as Dx, and the register derived from 0-2 is labelled as Dy. But for SBCD/PACK/UNPK/SUBX the labeling is swapped around. For each of these (except EXG) the assembly is consistently "MNEM src(0-2), dst(9-11)", so the behaviour is consistent but not the labeling.

In the SLEIGH specification, CMPM and EXG have been implemented separately, and PACK/UNPK are unimplemented, so this potentially only practically affects ABCD/SBCD/ADDX/SUBX. Currently the Tx/Ty tokens are consistent with ADDX/ABCD, derived from bits 9-11 and 0-2. Therefore, each instruction should use Tx as the destination and Ty as the source for both the display and the behaviour (pcode).
 
Prior to this suggested change, an assessment of each instruction is:
·         ADDX: this looks correct, as the pcode uses Tx as the destination, and the display is ordered Ty, Tx.
·         ABCD: the display looks correct, but the pcode uses Ty as the destination.
·         SUBX: the display and pcode look correct (but with labeling inconsistent with the manual).
·         SBCD: the display looks correct (but inconsistent with the manual), but the pcode uses Ty as the destination.

Therefore two changes are required: swapping Ty and Tx in the Pcode of ABCD and SBCD.

As your most recent update reflects exactly this for SBCD, I suggest just one more change to resolve the root cause: performing the same adjustment for ABCD.

**Comment by GhidorahRex** [comment]
@flk0 your analysis looks spot on. It would be helpful to add abcd to this same PR so we can close both at once.

**Comment by agatti** [comment]
@flk0 for what it's worth, #4223 adds PACK/UNPK - I had to switch operands' order around there as well.

**Comment by flk0** [comment]
@agatti Thanks. I have not tested this, but based on my previous comments I believe you need to amend your change by swapping X and Y, so that the second displayed register is the destination and is derived from bits 9-11. Again, this is caused by the inconsistent use of X and Y in the manual and the choice of X as the destination (bits 9-11) in the current SLEIGH implementation.

**Comment by agatti** [comment]
@flk0 indeed looks like you're right on that as well - I've updated the PR. Thanks!

**Pull request closed without merging**